.github/workflows/ci.yml

```diff
@@ -47,13 +47,47 @@ jobs:
         run: |
           go test ./... -short
   integration-tests:
+    strategy:
+      fail-fast: false
+      matrix:
+        git-version:
+          - 2.20.0 # oldest supported version
+          - 2.22.5
+          - 2.23.0
+          - 2.25.1
+          - 2.30.8
+          - latest
     runs-on: ubuntu-latest
-    name: "Integration Tests"
+    name: "Integration Tests - git ${{matrix.git-version}}"
     env:
       GOFLAGS: -mod=vendor
     steps:
       - name: Checkout code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
+      - name: Restore Git cache
+        if: matrix.git-version != 'latest'
```

Comment:
Is the idea that the latest git version is already installed? If so could you leave a comment to that effect?

Reply:
We know that *some* git version must already be installed, because before this branch we didn't bother installing git at all.

So this relies on the assumption that github's CI VMs are up to date wrt which git version they have. (I added the [print git version](https://github.com/jesseduffield/lazygit/pull/2754/commits/3c5e85eabeed4cee3d5a5436a456c08d91c0636d#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR104-R105) step specifically so that we can verify this.) At the moment it is the current one.

In an earlier iteration of the branch I had an entry in the matrix for `2.41.0` instead of `latest`; this simplifies the code a bit, as we don't need all the conditional logic for 'latest', but the downside is that we need to update the matrix every time a new git version is released, which I thought is too easy to forget. But I can easily revert back to that if you prefer it; let me know what you think.

Reply:
Sounds good